test_integration/commands_test.go

```diff
@@ -0,0 +1,98 @@
+package test_integration
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/bradleyjkemp/cupaloy"
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	setupCommands [][]string
+	actCommand    []string
+}
+
+const (
+	dbUser     = "postgres"
+	dbPort     = "5432"
+	dbName     = "test_integration"
+	dbHost     = "localhost"
+	dbPassword = "pass"
+)
+
+var mainFile = getMainFile()
+
+var testCases = map[string]testCase{
+	"policy-output": {
+		setupCommands: [][]string{
+			{"dropdb", "-h", dbHost, "-p", dbPort, "-U", dbUser, dbName},
+			{"createdb", "-h", dbHost, "-p", dbPort, "-U", dbUser, dbName},
+			// Created using `pg_dump -h 127.0.0.1 -U postgres test_integration > test_integration/fixtures/fetch_data/aws_s3.pgsql` after fetching only `s3*` resources, and sanitizing the output
+			{"psql", fmt.Sprintf("host=%s port=%s dbname=%s user=%s", dbHost, dbPort, dbName, dbUser), "-f", getFixtureFilePath("fetch_data/aws_s3.pgsql")},
+		},
+		actCommand: []string{"go", "run", mainFile, "policy", "run", "github.com/cloudquery-policies/aws?ref=v0.1.14", "--config", getFixtureFilePath("fetch_data/aws_config.hcl")},
```

Comment:
Using pinned versions of the provider (in `config.hcl`) and policy (in the argument) to keep the test stable, as we want to test only the output part